sopel/modules/emailcheck.py

```diff
@@ -0,0 +1,313 @@
+# coding=utf-8
+"""
+emailcheck.py - Watch oper messages for new nicks being registered
+Copyright © 2021, Kufat <kufat@kufat.net>
+Based on existing sopel code.
+Licensed under the Eiffel Forum License 2.
+"""
+
+import logging
+import re
+import urllib
+
+import sqlalchemy.sql
+
+from dataclasses import dataclass
+
+from sopel import db, module
+from sopel.config.types import FilenameAttribute, StaticSection, ValidatedAttribute, ListAttribute
+from sopel.tools import events, target, Identifier
+
+from sqlalchemy import Column, String, Float, Boolean, TIMESTAMP
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.ext.declarative import declarative_base
+
+try:
+    from ip import get_exemption
+    from ip import ipqs_lock
+except:
+    import threading
+
+    def get_exemption(ip):
+        return "Can't access exemptions; failing safe"
+
+    ipqs_lock = threading.Lock()
+
+
+EMAIL_REGEX = re.compile(r"([a-zA-Z0-9_.+-]+)@([a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+)")
+IRCCLOUD_USER_REGEX = re.compile(r"[us]id[0-9]{4,}")
+DOMAIN_LEN = 50
+
+DEFAULT_EXEMPT_SUFFIXES = {
```

Comment:
Getting rid of this. Easier to just allow gmail, etc. to be checked once and treated normally.